pkg/plugin/builtin/v1/internal/fromplugin/specifier.go

```diff
@@ -40,16 +58,37 @@ func SpecifierSpecifyResponse(in cpluginv1.SpecifierSpecifyResponse) (plugin.Spe
 }
 
 func SpecifierParameter(in cpluginv1.SpecifierParameter) plugin.Parameter {
-	var validations []plugin.Validation
-	if in.Required {
+	validations := make([]plugin.Validation, len(in.Validations))
+
+	requiredExists := false
+	for i, v := range in.Validations {
+		validations[i] = plugin.Validation{
+			Type:  plugin.ValidationType(v.Type),
+			Value: v.Value,
+		}
+		if v.Type == cpluginv1.ValidationTypeRequired {
+			requiredExists = true
+		}
+	}
+	// making sure not to duplicate the required validation
+	if in.Required && !requiredExists {
```

Comment:
I guess this exists for compatibility reasons?

Reply:
exactly

Reply:
Thanks for confirming! It would be nice to mention that in a comment.

Reply:
done!